linux/src/main/java/com/github/weisj/darkmode/platform/linux/gnome/GtkVariants.java

```diff
@@ -0,0 +1,123 @@
+package com.github.weisj.darkmode.platform.linux.gnome;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class is effectively a conversion of the GtkVariants.js module seen <a href="https://gitlab.com/rmnvgr/nightthemeswitcher-gnome-shell-extension/-/blob/0d32907de1d004a6a92f02b8e0b79302340e5244/src/modules/GtkVariants.js">here</a>.
+ * <p>This class should only be used as a temporary solution until settings are implemented which allow the user to
+ * denote which of their themes they consider to be their light and dark themes. This settings should be applicable to
+ * all distributions (not just ones using Gnome) as most distributions have different themes which provide a dark and
+ * light variant. Very few distributions will simply have a "dark <i>mode</i>" and a "light <i>mode</i>" like Windows
+ * and Mac OS.
+ */
+public class GtkVariants {
+    /**
+     * This documentation is straight from the file mentioned in the {@link GtkVariants} JavaDoc.
+     * <p/>
+     * The magic of guessing theme variants happens here.
+     * <p>
+     * If the theme doesn't fit a particular case, we'll do the following:
+     * <p> 	- Remove any signs of a dark variant to the theme name to get the day
+     * variant
+     * <p> 	- Remove any signs of a light variant to the day variant and add '-dark' to
+     * get the night variant
+     * <p><p>
+     * <p>For themes that don't work with the general rule, a particular case must be written. Day and night variants
+     * should be guessed with the most generic light and dark variants the theme offer, except if the user explicitly
+     * chose a specific variant.
+     * <p><p>
+     * <p> Light variants, from the most to the least generic:
+     * <p> 	- ''
+     * <p> 	- '-light'
+     * <p> 	- '-darker'
+     * <p>
+     * <p> Dark variants, from the most the least generic:
+     * <p> 	- '-dark'
+     * <p> 	- '-darkest'
+     *
+     * @param themeName the name of a theme. This could be one of the many variants of your theme (i.e. the dark version
+     *                  or whatever other variants the theme offers)
+     * @return a map containing three keys: "original" (the provided themeName), "day" (the guessed day variant), and
+     * "night" (the guessed night variant)
+     */
+    public static Map<String, String> guessFrom(String themeName) {
```

Comment:
This class would gain a lot of readability by migrating it into kotlin. Also please avoid stringly typed apis. `"night"` and `"day"` can be an enum.

Reply:
Fixed in 626360a51418c43b16ce28306f514cf75cbf0c5f and 945bb26422e83d4cd6db086d83c4af8eed504406.